### ISSUE TITLE
Remove deprecated twistd --report-profile

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4236]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4236|
|Type|task|
|Created|2010-01-27 17:27:08Z|
|Branch|https://github.com/twisted/twisted/tree/report-profile-4236-2|

[#2908](https://github.com/twisted/twisted/issues/2908) in 2007 deprecated the ```--report-profile``` of ```twistd``` so let's remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__4236 4236
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_report_profile_4236_2 branches/report-profile-4236-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__twistd twistd
time__1264613228000000 1264613228000000
changetime__1297572569000000 1297572569000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [28080]) Remove references to report-profile, refs [#4236](https://github.com/twisted/twisted/issues/4236)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [28081]) Pyflakes fixes, refs [#4236](https://github.com/twisted/twisted/issues/4236)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra removed owner|
|-|-|

Up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @thijstriemstra<br>@exarkun set status to `new`|
|-|-|

  1. Add a news file for the removal.
  1. Methods should be separated by two spaces.  In twisted/application/app.py, some methods that were no longer are.
  1. How did you make the `twisted/python/zsh/_twistd` change?  If I try to run zshcomp.py, it fails with an `ImportError`.  If I move it to another directory and invoke it with -d, it rewrites a lot of files, including _twistd, but it makes far more changes than are in the branch.

The rest looks fine, though.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [exarkun](#note_5):
>   1. How did you make the `twisted/python/zsh/_twistd` change?  If I try to run zshcomp.py, it fails with an `ImportError`.  If I move it to another directory and invoke it with -d, it rewrites a lot of files, including _twistd, but it makes far more changes than are in the branch.

I made those changes manually. How should I regenerate that _twistd file?

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

(In [30524]) apply coding standard, add news file. refs [#4236](https://github.com/twisted/twisted/issues/4236)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_10"></a>|@jerub set owner to @thijstriemstra|
|-|-|

There is a merge conflict in this branch due to the copyright year changing. You will need to fix this when merging.

Thank-you for doing this work, I think this branch should be merged.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

The merge conflict should be resolved by merging forward, of course.  Conflicts should never be resolved in the actual trunk merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [30709]) Branching to 'report-profile-4236-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [30711]) Merge report-profile-4236-2

Author: thijs
Reviewer: jerub, exarkun
Fixes: [#4236](https://github.com/twisted/twisted/issues/4236)

Remove the deprecated, long-since useless `--report-profile` twistd option.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
